source/common/quic/envoy_quic_server_stream.cc

```diff
@@ -306,5 +344,34 @@ QuicFilterManagerConnectionImpl* EnvoyQuicServerStream::filterManagerConnection(
   return dynamic_cast<QuicFilterManagerConnectionImpl*>(session());
 }
 
+HeaderValidationResult EnvoyQuicServerStream::validateHeader(const std::string& header_name,
+                                                             absl::string_view header_value) {
+  HeaderValidationResult result = EnvoyQuicStream::validateHeader(header_name, header_value);
+  if (result != HeaderValidationResult::ACCEPT) {
+    return result;
+  }
+  return checkHeaderNameForUnderscores(header_name);
+}
+
+HeaderValidationResult
+EnvoyQuicServerStream::checkHeaderNameForUnderscores(const std::string& header_name) {
```

Comment:
can we factor the Http2 code into HeaderUtility or not enough overlap?  or maybe at least stick it in the utility directory, unit terst, and we can use when we switch codec libraries.

Reply:
I moved this method to HeaderUtility, but using it in h2 codec doesn't save many lines, so I didn't use it there.